backend/api/public/attrs/crud.py

```diff
@@ -125,8 +126,9 @@ def filter_on_key_value_pairs(
     # If no filters applied, select all pulses
     if len(kv_pairs) == 0:
         pulses = db.exec(select(Pulse.pulse_id)).all()
-        if pulses[0] is None:
+        if not pulses:
```

Comment:
If `pulses` is a list of Nones, this will evaluate to false, which is wrong. I believe it was right before (but the return statement was missing).

```suggestion
        if pulses[0] is None:
```

Reply:
Well, I changed it back because I tested it. This is what happens if you test for `if pulses[0] is None]:

![CleanShot 2023-11-21 at 10 13 11@2x](https://github.com/GlazeTech/TeraStore/assets/15174457/86506c98-59e0-4a28-828c-e945be6b4f27)

The test is simple, no kv pair and no pulses uploaded:

![CleanShot 2023-11-21 at 10 11 33@2x](https://github.com/GlazeTech/TeraStore/assets/15174457/3397be1e-0d45-4ded-bf22-3adf20bcbfd1)

This hits this branch:

![CleanShot 2023-11-21 at 10 12 36@2x](https://github.com/GlazeTech/TeraStore/assets/15174457/5b44a986-ee65-4cc9-9304-8a1c461c8a42)

Reply:
Ah, so it should be extended to handle the case of no KV-pairs and no pulses. But what is no KV-pairs and there ARE pulses in the DB? Then it should return all pulses.

Reply:
> Ah, so it should be extended to handle the case of no KV-pairs and no pulses
Isn't that exactly what it does?

> But what is no KV-pairs and there ARE pulses in the DB? Then it should return all pulses.
The aneurysm I got from reading that aside, I guess you propose a new test? I'll get that in.

Reply:
Yes, that is exactly what it does, I meant it didn't do that on main - my bad. And yes, a test of that would be nice!